src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseExportService.java

```diff
@@ -170,13 +185,63 @@ private void exportProblemStatementAndEmbeddedFiles(ProgrammingExercise exercise
-        Set<String> embeddedFiles = new HashSet<>();
+        Set<String> embeddedFilesWithMarkdownSyntax = new HashSet<>();
+        Set<String> embeddedFilesWithHtmlSyntax = new HashSet<>();
+
+        Matcher matcherForMarkdownSyntax = Pattern.compile(EMBEDDED_FILE_MARKDOWN_SYNTAX_REGEX).matcher(exercise.getProblemStatement());
+        Matcher matcherForHtmlSyntax = Pattern.compile(EMBEDDED_FILE_HTML_SYNTAX_REGEX).matcher(exercise.getProblemStatement());
+        checkForMatchesInProblemStatementAndCreateDirectoryForFiles(outputDir, pathsToBeZipped, exportErrors, embeddedFilesWithMarkdownSyntax, matcherForMarkdownSyntax);
+        Path embeddedFilesDir = checkForMatchesInProblemStatementAndCreateDirectoryForFiles(outputDir, pathsToBeZipped, exportErrors, embeddedFilesWithHtmlSyntax,
+                matcherForHtmlSyntax);
+        // if the returned path is null the directory could not be created
+        if (embeddedFilesDir == null) {
+            return;
+        }
+        copyFilesEmbeddedWithMarkdownSyntax(exercise, exportErrors, embeddedFilesWithMarkdownSyntax, embeddedFilesDir);
+        copyFilesEmbeddedWithHtmlSyntax(exercise, exportErrors, embeddedFilesWithHtmlSyntax, embeddedFilesDir);
+
+    }
+
+    private void copyFilesEmbeddedWithHtmlSyntax(ProgrammingExercise exercise, List<String> exportErrors, Set<String> embeddedFilesWithHtmlSyntax, Path embeddedFilesDir) {
+        for (String embeddedFile : embeddedFilesWithHtmlSyntax) {
+            int indexOfFirstQuotationMark = embeddedFile.indexOf('"');
+            String filePath = embeddedFile.substring(embeddedFile.indexOf("src=") + 5, embeddedFile.indexOf('"', indexOfFirstQuotationMark + 1));
+            constructFilenameAndCopyFile(exercise, exportErrors, embeddedFilesDir, filePath);
+        }
+    }
+
+    private void constructFilenameAndCopyFile(ProgrammingExercise exercise, List<String> exportErrors, Path embeddedFilesDir, String filePath) {
+        String fileName = filePath.replace(API_MARKDOWN_FILE_PATH, "");
+        Path imageFilePath = Path.of(FilePathService.getMarkdownFilePath(), fileName);
+        Path imageExportPath = embeddedFilesDir.resolve(fileName);
+        // we need this check as it might be that the matched string is different and not filtered out above but the file is already copied
+        if (!Files.exists(imageExportPath)) {
+            try {
+                Files.copy(imageFilePath, imageExportPath);
+            }
+            catch (IOException e) {
+                exportErrors.add("Failed to copy embedded files: " + e.getMessage());
+                log.warn("Could not copy embedded file {} for exercise with id {}", fileName, exercise.getId());
+            }
+        }
+    }
+
+    private void copyFilesEmbeddedWithMarkdownSyntax(ProgrammingExercise exercise, List<String> exportErrors, Set<String> embeddedFilesWithMarkdownSyntax, Path embeddedFilesDir) {
```

Comment:
Can you swap this method with the one above? I think that order would make more sense